src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,1077 @@
+        with open(example_file1, 'w') as f:
+            f.write(json.dumps(sample_status_json))
+        self.runtime.status_handler.status_file_path = example_file1
+        self.runtime.status_handler.load_status_file_components(initial_load=True)
+
+        # Assert set_package_install_status
+        self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions, 'Installed', 'Critical')
+        with self.runtime.env_layer.file_system.open(self.runtime.status_handler.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"][0]
+
+        self.__assert_installation_set_packages_methods(substatus_file_data, Constants.PATCH_INSTALLATION_SUMMARY, patch_count)
+
+        # Assert set_package_install_status_classification
+        self.runtime.status_handler.set_package_install_status_classification(test_packages, test_package_versions, "Critical")
+        with self.runtime.env_layer.file_system.open(self.runtime.status_handler.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"][0]
+
+        self.__assert_installation_set_packages_methods(substatus_file_data, Constants.PATCH_INSTALLATION_SUMMARY, patch_count)
+
+    def test_only_assessment_patches_truncation_under_size_limit(self):
+        """ Perform no truncation on assessment patches.
+        Expecting:
+        assessment substatus status: success,
+        no assessment tombstone records,
+        assessment errors code: 0 (success),
+        assessment errors details code: 0 (success). """
+
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        patch_count = 500
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.__assert_patch_summary_from_status(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS)
+        self.__asert_message_json_from_status(substatus_file_data[0]["status"]["substatus"][0], patch_count, error_count=0)
+
```

Comment:
__assert_patch_summary_from_status and __asert_message_json_from_status have potential to be combined into one function

Reply:
done.